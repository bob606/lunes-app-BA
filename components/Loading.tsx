import React from 'react';
import {View, ActivityIndicator} from 'react-native';
import {COLORS} from '../constants/colors';
import {ILoadingProps} from '../interfaces';
import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  root: {
    flex: 1,
    alignItems: 'center',
  },
  indicator: {
    height: '50%',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
const Loading = ({children, isLoading}: ILoadingProps) => (
  <View style={styles.root}>
    {isLoading ? (
      <View style={styles.indicator}>
        <ActivityIndicator size="large" color={COLORS.lunesBlack} />
      </View>
    ) : (
      children
    )}
  </View>
);

export default Loading;
