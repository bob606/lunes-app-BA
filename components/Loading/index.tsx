import {
  React,
  styles,
  View,
  ActivityIndicator,
  COLORS,
  ILoadingProps,
} from './imports';

const Loading = ({children, isLoading}: ILoadingProps) => {
  return (
    <View style={styles.root}>
      {isLoading ? (
        <View style={styles.indicator}>
          <ActivityIndicator size="large" color={COLORS.lunesBlack} />
        </View>
      ) : (
        children
      )}
    </View>
  );
};

export default Loading;
