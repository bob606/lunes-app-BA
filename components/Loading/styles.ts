import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  root: {
    flex: 1,
    alignItems: 'center',
  },
  indicator: {
    height: '50%',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
