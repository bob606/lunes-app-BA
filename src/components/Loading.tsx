import React, { ReactElement } from 'react'
import { ActivityIndicator } from 'react-native'
import styled, { useTheme } from 'styled-components/native'

const Indicator = styled.View`
  height: 50%;
  justify-content: center;
  align-items: center;
`

interface ILoadingProps {
  children?: ReactElement
  isLoading: boolean
}

const Loading = ({ children, isLoading }: ILoadingProps): ReactElement => {
  const theme = useTheme()
  return (
    <>
      {isLoading ? (
        <Indicator>
          <ActivityIndicator size='large' color={theme.colors.lunesBlack} testID={'loading'} />
        </Indicator>
      ) : (
        children
      )}
    </>
  )
}

export default Loading
