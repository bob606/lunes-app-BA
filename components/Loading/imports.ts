import React from 'react';
import {View, ActivityIndicator} from 'react-native';
import {styles} from './styles';
import {COLORS} from '../../constants/colors';

export {React, styles, View, ActivityIndicator, COLORS};
