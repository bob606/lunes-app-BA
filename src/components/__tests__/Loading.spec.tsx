import { RenderAPI, render } from '@testing-library/react-native'
import React from 'react'
import { Text } from 'react-native'

import wrapWithTheme from '../../testing/wrapWithTheme'
import Loading from '../Loading'

describe('Loading', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const childText = 'Children'

  const renderLoading = (isLoading: boolean): RenderAPI =>
    render(
      <Loading isLoading={isLoading}>
        <Text>{childText}</Text>
      </Loading>,
      { wrapper: wrapWithTheme }
    )

  it('should not render children when isLoading is true', () => {
    const { queryByText, getByTestId } = renderLoading(true)
    expect(queryByText(childText)).toBeFalsy()
    expect(getByTestId('loading')).toBeTruthy()
  })

  it('should render children when isLoading is false', () => {
    const { getByText, queryByTestId } = renderLoading(false)
    expect(getByText(childText)).toBeTruthy()
    expect(queryByTestId('loading')).toBeFalsy()
  })
})
