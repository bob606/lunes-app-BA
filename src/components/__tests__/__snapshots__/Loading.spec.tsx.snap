// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Loading  renders correctly across screens 1`] = `
<Styled(View)>
  <Styled(View)>
    <ActivityIndicator
      animating={true}
      color="#000e38"
      hidesWhenStopped={true}
      size="large"
    />
  </Styled(View)>
</Styled(View)>
`;
